simphony/cuds/model.py

```diff
@@ -39,119 +55,179 @@ def __init__(self):
         # that will return the corresponding value.
         self._map = {}
 
-    @staticmethod
-    def _is_cuds_component(obj):
-        """Check whether the object is a CUDS component."""
-        # When I see a bird that walks like a duck and swims like a duck
-        # and quacks like a duck, I call that bird a duck
-        # TODO: replace with metadata type checks
-        return all([hasattr(obj, 'name'),
-                    hasattr(obj, 'data')])
+        # Another map to keep a mapping between names and ids
+        self._name_to_id_map = {}
+
+    @property
+    def uid(self):
+        return self._uid
+
+    @uid.setter
+    def uid(self, value):
```

Comment:
I agree, please remove the `setter` and `deleter`  readonly properties only need the `property` decorated function only